src/components/app-navigator/Header.tsx

```diff
@@ -0,0 +1,92 @@
+import { useCallback } from 'react'
+
```

Comment:
yeah these are great points, i was just going with the react-navigation language here, but topbar makes sense for consistency

Reply:
@sliptype i am also curious what the best move here is regarding context vs redux. it seems that different stacks can render different headers, so maybe the move is to let the screens define them? it might be a bit more direct and programatic that way. def need to look into what way is most standard and efficient tho

Reply:
That's a great idea, the current implementation isn't very declarative